### ISSUE TITLE
根据配置文件安装环境报错

### ISSUE THREAD

**Issue by liaogulou** [damo-vilab/videocomposer#15]
<img width="1137" alt="problem" src="https://github.com/damo-vilab/videocomposer/assets/113007768/fb91a25d-4911-423b-b432-221305997f19">

- 根据conda env create -f environment.yaml 安装环境报错，请问这个该如何解决呢？

**Comment by catherine-zky** [comment]
遇到了同样的问题，可以activate了环境之后用这个命令安装torch：pip install torch==1.12.0+cu113 torchvision==0.13.0+cu113 torchaudio==0.12.0 --extra-index-url https://download.pytorch.org/whl/cu113。可以在pytorch官网找到：https://pytorch.org/get-started/previous-versions/

**Comment by liaogulou** [comment]
> 
意思就是我要先手动安装一下torch，然后再去手动安装其他包吗？

**Comment by catherine-zky** [comment]
是的。activate新建的environment然后pip install所有的packages。flash-attn会花比较久的时间可以考虑单独装。conda activate VideoComposer 然后可以用下面这个command：
pip install absl-py==1.4.0 aiohttp==3.8.4 aiosignal==1.3.1 aliyun-python-sdk-core==2.13.36 aliyun-python-sdk-kms==2.16.0 asttokens==2.2.1 async-timeout==4.0.2 attrs==22.2.0 backcall==0.2.0 cachetools==5.3.0 cffi==1.15.1 chardet==5.1.0 charset-normalizer==3.1.0 clean-fid==0.1.35 click==8.1.3 cmake==3.26.0 crcmod==1.7 cryptography==39.0.2 decorator==5.1.1 decord==0.6.0 easydict==1.10 einops==0.6.0 executing==1.2.0 fairscale==0.4.6 filelock==3.10.2 flash-attn==0.2.0 frozenlist==1.3.3 fsspec==2023.3.0 ftfy==6.1.1 future==0.18.3 google-auth==2.16.2 google-auth-oauthlib==0.4.6 grpcio==1.51.3 huggingface-hub==0.13.3 idna==3.4 imageio==2.15.0 importlib-metadata==6.1.0 ipdb==0.13.13 ipython==8.11.0 jedi==0.18.2 jmespath==0.10.0 joblib==1.2.0 lazy-loader==0.2 markdown==3.4.3 markupsafe==2.1.2 matplotlib-inline==0.1.6 motion-vector-extractor==1.0.6 multidict==6.0.4 mypy-extensions==1.0.0 networkx==3.1 numpy==1.24.2 oauthlib==3.2.2 open-clip-torch==2.0.2 openai-clip==1.0.1 opencv-python==4.5.5.64 opencv-python-headless==4.7.0.68 oss2==2.17.0 packaging==23.0 parso==0.8.3 pexpect==4.8.0 pickleshare==0.7.5 pillow==9.4.0 pkgconfig==1.5.5 prompt-toolkit==3.0.38 protobuf==4.22.1 ptyprocess==0.7.0 pure-eval==0.2.2 pyasn1==0.4.8 pyasn1-modules==0.2.8 pycparser==2.21 pycryptodome==3.17 pydeprecate==0.3.1 pygments==2.14.0 pynvml==11.5.0 pyre-extensions==0.0.23 pytorch-lightning==1.4.2 pywavelets==1.4.1 pyyaml==6.0 regex==2023.3.22 requests==2.28.2 requests-oauthlib==1.3.1 rotary-embedding-torch==0.2.1 rsa==4.9 sacremoses==0.0.53 scikit-image==0.20.0 scikit-learn==1.2.2 scikit-video==1.1.11 scipy==1.9.1 simplejson==3.18.4 six==1.16.0 sklearn==0.0.post4 stack-data==0.6.2 tensorboard==2.12.0 tensorboard-data-server==0.7.0 tensorboard-plugin-wit==1.8.1 threadpoolctl==3.1.0 tifffile==2023.4.12 tokenizers==0.12.1 tomli==2.0.1 torch==1.12.0+cu113 torchaudio==0.12.0+cu113 torchmetrics==0.6.0 torchvision==0.13.0+cu113 tqdm==4.65.0 traitlets==5.9.0 transformers==4.18.0 triton==2.0.0.dev20221120 typing-extensions==4.5.0 typing-inspect==0.8.0 urllib3==1.26.15 wcwidth==0.2.6 werkzeug==2.2.3 xformers==0.0.13 yarl==1.8.2 zipp==3.15.0

**Comment by liaogulou** [comment]
> 是的。activate新建的environment之后先安装torch: pip install torch==1.12.0+cu113 torchvision==0.13.0+cu113 torchaudio==0.12.0 --extra-index-url https://download.pytorch.org/whl/cu113 然后pip install所有的packages。flash-attn会花比较久的时间可以考虑单独装。conda activate VideoComposer 然后可以用下面这个command： pip install absl-py==1.4.0 aiohttp==3.8.4 aiosignal==1.3.1 aliyun-python-sdk-core==2.13.36 aliyun-python-sdk-kms==2.16.0 asttokens==2.2.1 async-timeout==4.0.2 attrs==22.2.0 backcall==0.2.0 cachetools==5.3.0 cffi==1.15.1 chardet==5.1.0 charset-normalizer==3.1.0 clean-fid==0.1.35 click==8.1.3 cmake==3.26.0 crcmod==1.7 cryptography==39.0.2 decorator==5.1.1 decord==0.6.0 easydict==1.10 einops==0.6.0 executing==1.2.0 fairscale==0.4.6 filelock==3.10.2 flash-attn==0.2.0 frozenlist==1.3.3 fsspec==2023.3.0 ftfy==6.1.1 future==0.18.3 google-auth==2.16.2 google-auth-oauthlib==0.4.6 grpcio==1.51.3 huggingface-hub==0.13.3 idna==3.4 imageio==2.15.0 importlib-metadata==6.1.0 ipdb==0.13.13 ipython==8.11.0 jedi==0.18.2 jmespath==0.10.0 joblib==1.2.0 lazy-loader==0.2 markdown==3.4.3 markupsafe==2.1.2 matplotlib-inline==0.1.6 motion-vector-extractor==1.0.6 multidict==6.0.4 mypy-extensions==1.0.0 networkx==3.1 numpy==1.24.2 oauthlib==3.2.2 open-clip-torch==2.0.2 openai-clip==1.0.1 opencv-python==4.5.5.64 opencv-python-headless==4.7.0.68 oss2==2.17.0 packaging==23.0 parso==0.8.3 pexpect==4.8.0 pickleshare==0.7.5 pillow==9.4.0 pkgconfig==1.5.5 prompt-toolkit==3.0.38 protobuf==4.22.1 ptyprocess==0.7.0 pure-eval==0.2.2 pyasn1==0.4.8 pyasn1-modules==0.2.8 pycparser==2.21 pycryptodome==3.17 pydeprecate==0.3.1 pygments==2.14.0 pynvml==11.5.0 pyre-extensions==0.0.23 pytorch-lightning==1.4.2 pywavelets==1.4.1 pyyaml==6.0 regex==2023.3.22 requests==2.28.2 requests-oauthlib==1.3.1 rotary-embedding-torch==0.2.1 rsa==4.9 sacremoses==0.0.53 scikit-image==0.20.0 scikit-learn==1.2.2 scikit-video==1.1.11 scipy==1.9.1 simplejson==3.18.4 six==1.16.0 sklearn==0.0.post4 stack-data==0.6.2 tensorboard==2.12.0 tensorboard-data-server==0.7.0 tensorboard-plugin-wit==1.8.1 threadpoolctl==3.1.0 tifffile==2023.4.12 tokenizers==0.12.1 tomli==2.0.1 torch==1.12.0+cu113 torchaudio==0.12.0+cu113 torchmetrics==0.6.0 torchvision==0.13.0+cu113 tqdm==4.65.0 traitlets==5.9.0 transformers==4.18.0 triton==2.0.0.dev20221120 typing-extensions==4.5.0 typing-inspect==0.8.0 urllib3==1.26.15 wcwidth==0.2.6 werkzeug==2.2.3 xformers==0.0.13 yarl==1.8.2 zipp==3.15.0

哈哈，感觉你在环境这个上面折磨了很久，那环境安装完成之后，你有没有遇到这个issue[问题](https://github.com/damo-vilab/videocomposer/issues/14)啊，CUDA error: no kernel image is available for execution on the device

**Comment by catherine-zky** [comment]
没有遇到cuda的问题。我用的40G的gpu。

**Comment by YukunZEX** [comment]
@liaogulou 请问你按这个方法安装解决了么？

**Comment by Never-up** [comment]
> @liaogulou 请问你按这个方法安装解决了么？

解决啦

**Comment by suimuc** [comment]
吐槽一下这个项目开源开的真的依托答辩
